backend/tests/Feature/NotificationTest.php

```diff
@@ -227,4 +227,92 @@ public function testNotificationsCanOnlyBeQueriedForOneself()
         ];
         $response->assertJson($expected_data);
     }
+
+    /**
+     * @return void
+     */
+    public function testMarkNotificationReadViaGraphqlEndpoint()
+    {
+        /** @var User $user */
+        $user = User::factory()->create();
+        $this->actingAs($user);
+        $notification_data = $this->getSampleNotificationData($user, 1010);
+        $user->notify(new SubmissionCreated($notification_data));
+        $this->graphQL(
+            'mutation MarkNotificationRead ($notification_id: ID!) {
+                markNotificationRead(id: $notification_id) {
+                  read_at
+                }
+            }',
+            [
+                'notification_id' => $user->notifications->first()->id,
+            ]
+        );
+        $this->assertEquals(0, $user->unreadNotifications()->count());
+    }
+
+    /**
+     * @return void
+     */
+    public function testMarkNotificationReadViaGraphqlEndpointIsProhibitedAsAnotherUser()
+    {
+        /** @var User $user_1 */
+        $user_1 = User::factory()->create();
+        $user_2 = User::factory()->create();
+        $this->actingAs($user_1);
+        $notification_data = $this->getSampleNotificationData($user_2, 1011);
+        $user_2->notify(new SubmissionCreated($notification_data));
+        $this->graphQL(
+            'mutation MarkNotificationRead ($notification_id: ID!) {
+                markNotificationRead(id: $notification_id) {
+                  read_at
+                }
+            }',
+            [
+                'notification_id' => $user_2->notifications->first()->id,
+            ]
+        );
+        $this->assertEquals(1, $user_2->unreadNotifications()->count());
+    }
+
+    /**
+     * @return void
+     */
+    public function testMarkAllNotificationsReadViaGraphqlEndpoint()
+    {
+        /** @var User $user */
+        $user = User::factory()->create();
+        $this->actingAs($user);
+        $notification_data_1 = $this->getSampleNotificationData($user, 1012);
+        $user->notify(new SubmissionCreated($notification_data_1));
+        $notification_data_2 = $this->getSampleNotificationData($user, 1013);
+        $user->notify(new SubmissionCreated($notification_data_2));
+        $this->graphQL(
+            'mutation MarkAllNotificationsRead {
+                markAllNotificationsRead
+            }'
+        );
+        $this->assertEquals(0, $user->unreadNotifications()->count());
```

Comment:
Probably should assert the return value of the mutation here as well (count of notifications marked as read)

Reply:
Added